low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -410,77 +440,94 @@ def _generate_host_update(self):
                 self._host_memory_usage_threshold = float(self._host_memory_usage_threshold)
         except ValueError:
             logger.warning("Host Memory Alert, Invalid host_memory_usage_threshold value are entered in config.")
-            # Assigning default value to _disk_usage_threshold
+            # Assigning default value to _memory_usage_threshold
             self._host_memory_usage_threshold = self.DEFAULT_HOST_MEMORY_USAGE_THRESHOLD
+        
+        memory_persistent_data = self.read_persistent_data('MEMORY_USAGE_DATA')
+        if memory_persistent_data['memory_usage_time_map']:
+            previous_check_time = int(memory_persistent_data['memory_usage_time_map'])
+        else:
+            previous_check_time = int(-1)

```

Comment:
why not assign previous_check_time  with current time here itself?

Reply:
So that we can restart the checking process if fault is resolved. When it gets the usage higher than the threshold for the first time then only we want previous_check_time to update. So that every subsequent time when it gets usage higher than threshold, it subtract the current time with the time when it got the above threshold usage for the first time (which is previous_check_time)

Reply:
>When it gets the usage higher than the threshold for the first time then only we want previous_check_time to update.

This suffice, thanks.